app/javascript/mastodon/components/avatar_overlay.jsx

```diff
@@ -1,51 +0,0 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import ImmutablePropTypes from 'react-immutable-proptypes';
-import { autoPlayGif } from '../initial_state';
-import Avatar from './avatar';
-
-export default class AvatarOverlay extends React.PureComponent {
-
-  static propTypes = {
-    account: ImmutablePropTypes.map.isRequired,
-    friend: ImmutablePropTypes.map.isRequired,
-    animate: PropTypes.bool,
-    size: PropTypes.number,
-    baseSize: PropTypes.number,
-    overlaySize: PropTypes.number,
-  };
-
-  static defaultProps = {
-    animate: autoPlayGif,
-    size: 46,
-    baseSize: 36,
-    overlaySize: 24,
-  };
-
-  state = {
-    hovering: false,
-  };
-
-  handleMouseEnter = () => {
-    if (this.props.animate) return;
-    this.setState({ hovering: true });
-  };
-
-  handleMouseLeave = () => {
-    if (this.props.animate) return;
-    this.setState({ hovering: false });
-  };
-
-  render() {
-    const { account, friend, animate, size, baseSize, overlaySize } = this.props;
-    const { hovering } = this.state;
-
-    return (
-      <div className='account__avatar-overlay' style={{ width: size, height: size }}>
-        <div className='account__avatar-overlay-base'><Avatar animate={hovering || animate} account={account} size={baseSize} /></div>
```

Comment:
No need to pass animate arguments  to Avatar component why they have hovering state and its event handler in themself.

Reply:
Looks like the event handlers were not hooked up but I think the intent was that both avatars would play at the same time when the overlay was hovered.

Reply:
> Looks like the event handlers were not hooked up but I think the intent was that both avatars would play at the same time when the overlay was hovered.

I seem so too.
But this PR is a refactor, so I didn't change the behavior.
Should I fix it now?

This movie is behavior of current main branch(e5c0b1673).
https://user-images.githubusercontent.com/24884114/232274621-b9cca042-1533-452f-8f65-ee4fd2e504f3.mp4

Reply:
I made a commit.
https://github.com/mastodon/mastodon/pull/24543/commits/d770fd24b6c8f2d4dc6bd031be4b3a0e86c77e6e

https://user-images.githubusercontent.com/24884114/232283584-52e1290b-c44b-44f8-8d6e-544ec1bd5952.mp4